modules/covid_vaccine/app/services/covid_vaccine/v0/facility_lookup_service.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+require 'lighthouse/facilities/client'
+
+module CovidVaccine
+  module V0
+    class FacilityLookupService
+      # List of VA "consolidated" facilities, VAMCs that had their
+      # VistA instances merged and therefore are VAMCs that may have a
+      # longer than 3 digit station ID
+      CONSOLIDATED_FACILITIES = %w[528 528A4 528A5 528A6 528A7 528A8
+                                   549 589 589A4 589A5 589A6 589A7
+                                   620 620A4 626 626A4 636 636A6
+                                   636A8 657 657A4 657A5].freeze
+
+      # return a map of attributes including the zipcodes lat/long
+      # and the list of n closest health facilities
+      def facilities_for(zipcode)
+        zcta_row = ZCTA[zipcode&.[](0...5)]
+        return {} if zcta_row.blank?
+
+        lat = zcta_row[ZCTA_LAT_HEADER]
+        lng = zcta_row[ZCTA_LON_HEADER]
+        {
+          zip_code: zipcode,
+          zip_lat: lat,
+          zip_lon: lng
+        }.merge(nearest_facilities(lat, lng))
+      end
+
+      private
+
+      ## Get the n nearest facility IDs to the provided latitude/longitude
+      # Attempts to find n nearest by drive-time. If an insufficient number
+      # of results is returned based on limitations of the drive-time API,
+      # scraps those results and falls back to find the n nearest by distance
+      #
+      def nearest_facilities(lat, lng)
+        client = Lighthouse::Facilities::Client.new
+        response = client.nearby(lat: lat, lng: lng)
+        # Work around a bug in /nearby API that returns non-VHA facilities
+        response = response.filter { |x| x.id.start_with?('vha_') }
+        result = nearest_vamc(response.map { |x| x.id.delete_prefix('vha_') })
+        if result.blank?
+          # Does not seem feasible that a location would be closer to
+          # 30 clinics than any VAMCs
+          response = client.get_facilities(lat: lat, long: lng,
+                                           per_page: 30, type: 'health')
+          result = nearest_vamc(response.map { |x| x.id.delete_prefix('vha_') })
+        end
+        sta3n = result.last if result.last.length == 3
+        sta6a = result.first if result.first.length > 3
+        sta6a = result.last if result.last.length > 3
```

Comment:
Just making sure, if result.first.length > 3 and result.last.length > 3 are both true, sta6a should be set to result.last correct?

Reply:
Yes...tried to describe the logic but that means that the nearest VAMC is one of the "consolidated facilities" with a > 3 length station ID, so it will be both first and last in the list, and need to go into the sta6a field.